app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/parking_charge/AddParkingChargeForm.kt

```diff
@@ -0,0 +1,196 @@
+                        )
+                        TimeUnitButton(
+                            text = getString(R.string.quest_parking_charge_per_day),
+                            selected = timeUnitState.value == TimeUnit.DAY,
+                            onClick = {
+                                timeUnitState.value = TimeUnit.DAY
+                                checkIsFormComplete()
+                            },
+                            modifier = Modifier.weight(1f)
+                        )
+                    }
+
+                    // Auto-detected currency display
+                    Text(
+                        text = getString(R.string.quest_parking_charge_currency_info, getCurrencyForCountry()),
+                        style = MaterialTheme.typography.caption,
+                        color = MaterialTheme.colors.onSurface.copy(alpha = 0.6f)
+                    )
+                }
+
+                // Dialogue for "It varies"
+                if (showDialogState.value) {
+                    TextInputDialog(
+                        onDismissRequest = { showDialogState.value = false },
+                        onConfirmed = { description ->
+                            applyAnswer(ItVaries(description))
+                        },
+                        title = { Text(getString(R.string.quest_parking_charge_varies_title)) },
+                        textInputLabel = { Text(getString(R.string.quest_parking_charge_varies_description)) }
+                    )
+                }
+            }
+        }
+    }
+
+    override fun isFormComplete(): Boolean {
+        val amount = amountState.value.replace(',', '.')
+        return amount.isNotEmpty() && amount.toDoubleOrNull() != null && amount.toDouble() > 0
+    }
+
+    override fun onClickOk() {
+        val amount = amountState.value.replace(',', '.')
+        val currency = getCurrencyForCountry()
+        val timeUnit = when (timeUnitState.value) {
+            TimeUnit.HOUR -> "hour"
+            TimeUnit.DAY -> "day"
+        }
+
+        applyAnswer(SimpleCharge(amount, currency, timeUnit))
+    }
+
+    /** This function should probably be improved in the Future because it only matches some countries. */
+    private fun getCurrencyForCountry(): String = when (countryInfo.countryCode) {
+            "AT", "BE", "CY", "DE", "EE", "ES", "FI", "FR", "GR", "IE",
+            "IT", "LT", "LU", "LV", "MT", "NL", "PT", "SI", "SK" -> "EUR"
+            "GB" -> "GBP"
+            "US" -> "USD"
+            "CH" -> "CHF"
+            "DK" -> "DKK"
+            "SE" -> "SEK"
+            "NO" -> "NOK"
+            "PL" -> "PLN"
+            "CZ" -> "CZK"
+            "HU" -> "HUF"
+            "RO" -> "RON"
+            "BG" -> "BGN"
+            "HR" -> "HRK"
+            "JP" -> "JPY"
+            "CN" -> "CNY"
+            "IN" -> "INR"
+            "AU" -> "AUD"
+            "CA" -> "CAD"
+            else -> "EUR"
+        }
+}
```

Comment:
We actually collect such meta-information in a separate repo:

https://github.com/streetcomplete/countrymetadata

In StreetComplete, we then parse this information via the `CountryInfo` class.

Also, this is apparently https://en.wikipedia.org/wiki/ISO_4217#Active_codes , so maybe add a comment this.

So, it would be good if you created a PR that adds a file like `data/currency.yml` in that repo.